### PR TITLE
keyword rules for --doc only

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- based on https://github.com/Samsung/CredSweeper/pull/433
- added rules for --doc only 
- some new samples are detected with existed rules

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark - should be not touched due ``--doc`` is not used in benchmark
- [x] TODO: remove temporally version changing - one test failed

### CODE REVIEW

File: credsweeper/rules/config.yaml [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+  values:
+    - (?P<ddash>--)?(?P<variable>\w*(?i:pa[as]swords?|pass(wd)?|pwd|pin|\bp/w|\bpw|비밀번호|비번|패스워드|암호))\s*(?(ddash)[ =]|[:=/>-]{1,2})\s*(?P<quote>[`'\"]+)?(?P<value>\S{3,}?)(?(quote)(?P=quote)|\b)
+    - (?P<ddash>--)?(?i:id|login|account|root|admin|user|name|wifi|role|host|default|계정|아이디)\s*(?(ddash)[ =]|[ :=])\s*(?P<value>\S+)
+  filter_type:
+    - ValueAllowlistCheck
+  min_line_len: 10
+  required_substrings:
+    - sword
+    - pass
+    - pwd
```
Comment (babenek) [inline credsweeper/rules/config.yaml:79]:
```suggestion
```

Comment (babenek) [reply credsweeper/rules/config.yaml:79]: [no ci] removed extra

File: credsweeper/__init__.py [hunk 2]
```diff
@@ -20,4 +20,4 @@
     '__version__'
 ]
 
-__version__ = "1.5.7"
+__version__ = "1.5.7.20231017"
```
Comment (babenek) [inline credsweeper/__init__.py:23]:
```suggestion
__version__ = "1.5.7"
```

File: credsweeper/__init__.py [hunk 3]
```diff
@@ -20,4 +20,4 @@
     '__version__'
 ]
 
-__version__ = "1.5.7"
+__version__ = "1.5.7.20231019"
```
Comment (babenek) [inline credsweeper/__init__.py:23]:
```suggestion
__version__ = "1.5.7"
```

File: tests/test_app.py [hunk 4]
```diff
@@ -320,7 +320,8 @@ def test_version_p(self) -> None:
     def test_banner_p(self) -> None:
         _stdout, _stderr = self._m_credsweeper(["--banner"])
         output = " ".join(_stdout.split())
-        self.assertRegex(output, r"CredSweeper \d+\.\d+\.\d+ crc32:[0-9a-f]{8}", _stderr or _stdout)
+        # (.20[0-9]{6})? is optional - date of test commit
+        self.assertRegex(output, r"CredSweeper \d+\.\d+\.\d+(.20[0-9]{6})? crc32:[0-9a-f]{8}", _stderr or _stdout)
```
Comment (babenek) [inline tests/test_app.py:324]:
```suggestion
        self.assertRegex(output, r"CredSweeper \d+\.\d+\.\d+ crc32:[0-9a-f]{8}", _stderr or _stdout)
```

**Comment by babenek** [comment]
TODO: remove temporally versionisation with datetime (?)

File: tests/test_app.py [hunk 5]
```diff
@@ -604,6 +604,12 @@ def test_rules_ml_p(self) -> None:
             rules = Util.yaml_load(APP_PATH / "rules" / "config.yaml")
             rules_set = set([i["name"] for i in rules])
             missed = {  #
+                "SINGLE_STR_PAIR",
```
Comment (babenek) [inline tests/test_app.py:607]:
```suggestion
```

File: tests/test_app.py [hunk 6]
```diff
@@ -627,7 +633,15 @@ def test_rules_ml_n(self) -> None:
             report_set = set([i["rule"] for i in report])
             rules = Util.yaml_load(APP_PATH / "rules" / "config.yaml")
             rules_set = set([i["name"] for i in rules])
-            self.assertSetEqual(rules_set, report_set, f"\n{_stdout}")
+            missed = {  #
+                "SINGLE_STR_PAIR",
```
Comment (babenek) [inline tests/test_app.py:637]:
```suggestion
```

**Review by babenek — COMMENTED** [review]
tests  updated

**Review by csh519 — APPROVED** [review]
Unstructured pattern rules are covered by this PR.
We should improve more about this but it's okay for now.

LGTM :+1:

**Review by iuriimet — APPROVED** [review]

**Pull request merged**